pkg/certificate/mrc_reconciler.go

```diff
@@ -152,6 +153,81 @@ func (m *Manager) updateMRCComponentStatus(mrc *v1alpha2.MeshRootCertificate) er
 
 	return m.mrcClient.UpdateMeshRootCertificate(mrc)
 }
+func getNextStatus(mrc *v1alpha2.MeshRootCertificate) v1alpha2.MeshRootCertificateComponentStatus {
+	// possible status transitions:
+	// unknown -> validating
+	// validating -> issuing
+	// validating -> unused
+	// issuing -> unused
+	switch {
+	case shouldChangeStatusToValidating(mrc):
+		return constants.MRCComponentStatusValidating
+	case shouldChangeStatusToIssuing(mrc):
+		return constants.MRCComponentStatusIssuing
+	case shouldChangeStatusToUnused(mrc):
+		return constants.MRCComponentStatusUnused
+	default:
+		return constants.MRCComponentStatusUnknown
+	}
+}
+
+func shouldChangeStatusToUnused(mrc *v1alpha2.MeshRootCertificate) bool {
+	// TODO: implement check from issuing/validating to rollback
+	return false
+}
+
+func shouldChangeStatusToIssuing(mrc *v1alpha2.MeshRootCertificate) bool {
+	// TODO: implement check from validating to issuing
+	return false
+}
+
+// checks if the given MRC meets the attributes to move the componentStatuses to Validating
+func shouldChangeStatusToValidating(mrc *v1alpha2.MeshRootCertificate) bool {
+	// checks if intent is passive
+	if !strings.EqualFold(string(mrc.Spec.Intent), constants.MRCIntentPassive) {
+		return false
+	}
+	// checks if Accepted condition has status true
+	if !isConditionTrue(mrc, constants.MRCConditionTypeAccepted) {
+		return false
+	}
+
+	// checks if ValidatingRollout condition has status false with reason Pending
+	if isConditionFalse(mrc, constants.MRCConditionTypeValidatingRollout) && isReasonPending(mrc, constants.MRCConditionTypeValidatingRollout) {
```

Comment:
Should this be the same reason after https://github.com/openservicemesh/osm/pull/5183/files#diff-ee5985129a778d4127e4cf7da9b939021e6e3e7d78106a950391d3a0b96b793eR208?  (I think this two PR are working in the same area?)

I am not sure I understand why we need to do this in two phases.  After it is added the the cert validated, shouldn't we be able to mark it as Validating in the linked PR?  

I think this adds another event change to https://github.com/openservicemesh/osm/pull/5183/files#r988421071  Off of a single submission to the user? 

Since this is done by _each_ osm-controller pod, the first one done could kick off this set of events and another one could still be processing but we are reporting to the user that things are in a good state that isn't true for all the osm-controllers?